yazi-fm/src/components/manager.rs

```diff
@@ -18,3 +30,85 @@ impl Widget for Manager {
 		}
 	}
 }
+
+impl Manager {
+	pub fn mouse(
+		self,
+		event: crossterm::event::MouseEvent,
+		layout: &Layout,
+		cx: &crate::context::Ctx,
+	) {
+		let position = Position { x: event.column, y: event.row };
+		let comp = if layout.preview.contains(position) {
```

Comment:
The `Manager` component should only include the logic for `Moved` and `Drag(MouseButton)`, all others should be dispatched to specific components (`Parent`, `Current`, `Preview`). There are no `parent.rs` and `current.rs` currently, please create them.